BSP/Simulator_V2/Src/BSP_CAN.c

```diff
@@ -85,6 +104,9 @@ ErrorStatus BSP_CAN_Write(uint32_t id, uint8_t data[8], uint8_t length) {
  * @return  ERROR if nothing was received so ignore id and data that was received. SUCCESS indicates data was received and stored.
  */
 ErrorStatus BSP_CAN_Read(uint32_t *id, uint8_t *data) {
+    char buffer[50];
```

Comment:
It looks like it's for sprintf, not for holding the raw CAN data

Reply:
Clark is right, it's a char buffer for the entire message to print, not just the input. Regardless, CAN_READ doesn't have much functionality for some reason (Clark knows more about this).

Reply:
Yea, but we plan on using CAN_READ later on so we will have to get it working sometime.

Reply:
We'd have to use the FIFO and actually write messages into it I think, in that case. Also I have some things to say about the file structure for the sim later at today's meeting

Reply:
We'll have to add support to CAN to the JSON if we want to read it. I don't think it will be too bad, but it will be more complicated than just adding a field (otherwise having a lot of CAN messages will force you to have a ton of simulator states).

It wasn't a priority for me since it doesn't work on the hardware